src/python/pants/util/objects.py

```diff
@@ -50,7 +51,6 @@ def datatype(field_decls, superclass_name=None, **kwargs):
 
   if not superclass_name:
     superclass_name = '_anonymous_namedtuple_subclass'
-  superclass_name = str(superclass_name)
 
   namedtuple_cls = namedtuple(superclass_name, field_names, **kwargs)
```

Comment:
Thanks for fixing this -- when I wrote that I didn't understand the implications of `unicode_literals` and was having a lot of issues. There are multiple places I've used `datatype` with a field that should have been string-typed, and failed to add a type check because I wasn't sure how to handle it -- should have bubbled up the concern then. Will try to keep an eye on this in the future, but any further strangeness you may find with datatypes especially around string handling could be accidental, so please feel free to make these sorts of changes in the future. Thanks!

Reply:
It’s tricky code! I was thrown off by ABCMeta requiring bytes in python 2. 

I’ll continue to break out any changes we need to make due to `meta` in separate PRs.